### ISSUE TITLE
[Suggestion] Add XP for voice chat

### ISSUE THREAD

**Issue by 10by10pixel** [cookkkie/mee6#68]
Pretty simple. Just how it works for text chat. If a person speaks at any time within a minute, give the same amount of XP that you would for text chat.

**Comment by KazeSenoue** [comment]
Bots can't see voice activity on voice channels they're not in.

**Comment by LJNeon** [comment]
Actually they could get the voicestates of everyone in the server and add XP if they're talking regardless of if the bot is in a voice channel or not, but that's probably not a good idea.

**Comment by TattooNationGamersClub** [comment]
Sounds good, though.

On January 12, 2017, at 8:33 PM, LJNeon <notifications@github.com> wrote:

Actually they could get the voicestates of everyone in the server and add XP if they're talking regardless of if the bot is in a voice channel or not, but that's probably not a good idea.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.￼

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/cookkkie/mee6","title":"cookkkie/mee6","subtitle":"GitHub repository","main_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png">https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png</a>","avatar_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png">https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/cookkkie/mee6">https://github.com/cookkkie/mee6</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@LJNeon in #68: Actually they could get the voicestates of everyone in the server and add XP if they're talking regardless of if the bot is in a voice channel or not, but that's probably not a good idea."}],"action":{"name":"View Issue","url":"<a href="https://github.com/cookkkie/mee6/issues/68#issuecomment-272345446">https://github.com/cookkkie/mee6/issues/68#issuecomment-272345446</a>"}}}

**Comment by LJNeon** [comment]
Yes but for large servers it would be... what's the phrase... a costly function?

**Comment by 10by10pixel** [comment]
It's not that costly is it? It only runs once a minute...

**Comment by LJNeon** [comment]
Maybe, but you gotta understand how many servers and how many people in all those servers Mee6 is in, looping through every single person in general is a bad idea.

**Comment by 10by10pixel** [comment]
Wouldn't it be looping through every voice channel? Most servers would have only 3 or 4 and they're usually empty...
